src/index.ts

```diff
@@ -1,70 +1,84 @@
 import debounce from 'lodash/debounce';
 import throttle from 'lodash/throttle';
 
+type EventType = keyof GlobalEventHandlersEventMap;
+
+type EventData = {
+  listeners: Record<string, HandlerFunction>;
+  lastIndex: number;
+  listenersLength: number;
+};
+
+type EventSettings = {
+  emitter?: Document | typeof globalThis;
+  wrapper?: (callback: Function) => Function;
+};
+
+type HandlerFunction = Function & { _attachedAt: number, cancel?:Function };
+
 const RESIZE_RATE = 300;
 const SCROLL_RATE = 100;
 const HANDLER_CALL_DELAY = 100;
 
-// ========================================================================================
-// Initialization
-// ========================================================================================
 const isDOMAvailable = typeof window !== 'undefined';
 
-const settingsMap = {};
-const eventMap = {};
+const settingsMap: Record<string, EventSettings> = {};
+const eventMap: Record<string, EventData> = {};
 const noop = () => {};
-const defaultSettings = {};
-
+const defaultSettings: EventSettings = {};
 
 const createEventSettings = () => {
   settingsMap.resize = {
-    emitter: global,
-    wrapper(callback) { return debounce(callback, RESIZE_RATE); },
+    emitter: globalThis,
+    wrapper(callback) { return debounce(callback as any, RESIZE_RATE); },
```

Comment:
This won't be necessary if the function type is properly described and the other `anys`